### PR TITLE
[common] Introduced CudaStream an RAII class to wrap cuda streams

### LINKED ISSUE #393: [common] Wrapping CUDA stream in another class to catch stream destruction

### PR DESCRIPTION
**Pull request by ahehn-nv**
fixes #393

### CODE REVIEW

File: pygenomeworks/genomeworks/cuda/cuda.pyx [hunk 1]
```diff
@@ -32,40 +33,19 @@ cdef class CudaStream:
...
-            raise CudaRuntimeError(e)
-        self.stream = <size_t>s
+    cdef cuda_stream.CudaStream stream
 
     def __init__(self):
-        """Dummy implementation of __init__ function to allow
-        for Python subclassing.
+        """Constructs a CudaStream object to encapsulate a CUDA stream.
         """
+        self.stream = cuda_stream.make_cuda_stream()
```
Comment (tijyojwad) [inline pygenomeworks/genomeworks/cuda/cuda.pyx:42]: this needs to be in the `__cinit__` function. That's when all C objects need to be initialized

an empty `__init__ object also needs to be kept around to allow for python subclassing

File: pygenomeworks/genomeworks/cudaaligner/cudaaligner.pyx [hunk 2]
```diff
@@ -122,7 +122,6 @@ cdef class CudaAlignerBatch:
     """Python API for CUDA-accelerated sequence to sequence alignment.
     """
     cdef unique_ptr[cudaaligner.Aligner] aligner
-    cdef public object stream
```
Comment (tijyojwad) [inline pygenomeworks/genomeworks/cudaaligner/cudaaligner.pyx:125]: why was this removed? I think there's a reference to this below?

Comment (ahehn-nv) [reply pygenomeworks/genomeworks/cudaaligner/cudaaligner.pyx:125]: Because I would like this class to use the python object, not the c++ object.

Comment (tijyojwad) [reply pygenomeworks/genomeworks/cudaaligner/cudaaligner.pyx:125]: Does the `self.stream = stream` line in `__cinit__` work if the `stream` variable is not `cdef`ed with some type?

File: pygenomeworks/genomeworks/cuda/cuda.pyx (hunk 1 above)
Comment (ahehn-nv) [reply pygenomeworks/genomeworks/cuda/cuda.pyx]: I removed all the python changes from this PR.

File: pygenomeworks/genomeworks/cudaaligner/cudaaligner.pyx (hunk 2 above)
Comment (ahehn-nv) [reply pygenomeworks/genomeworks/cudaaligner/cudaaligner.pyx]: I removed all the python changes from this PR.

**Review by mimaric — APPROVED** [review]
